### PR TITLE
[QuickLookUI] Update bindings for Xcode 13.0 beta 1

### PR DESCRIPTION
**Pull request by rachelkang**
No updates for beta 2,3,4

related: https://github.com/xamarin/xamarin-macios/pull/12337, https://github.com/xamarin/xamarin-macios/issues/12388

### CODE REVIEW

File: src/quicklookUI.cs [hunk 1]
```diff
@@ -172,5 +173,69 @@ interface QLPreviewingController {
...
 		[Export ("preparePreviewOfFileAtURL:completionHandler:")]
 		void PreparePreviewOfFile (NSUrl url, Action<NSError> completionHandler);
+
+		[iOS (15,0), Mac (12,0), MacCatalyst (15,0)]
+		[Export ("providePreviewForFileRequest:completionHandler:")]
+		void ProvidePreview (QLFilePreviewRequest request, Action<QLPreviewReply, NSError> handler);
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
```
Comment (spouliot) [inline src/quicklookUI.cs:183]: add `[DisableDefaultCtor]`

the API seems to be used only inside a protocol and that means the instance is created by the caller (OS) and not by user code

Another hint is that there's no way to set the` FileUrl` property which makes user-created instance not _really_ usable.

File: src/quicklookUI.cs [hunk 2]
```diff
@@ -172,5 +173,69 @@ interface QLPreviewingController {
...
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface QLPreviewProvider : NSExtensionRequestHandling
+	{
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface QLPreviewReplyAttachment
```
Comment (spouliot) [inline src/quicklookUI.cs:199]: add `[DisableDefaultCtor]`

there's a `init*` method and no setters on the properties

File: src/quicklookUI.cs [hunk 3]
```diff
@@ -172,5 +173,69 @@ interface QLPreviewingController {
...
+	delegate bool QLPreviewReplyDrawingHandler (CGContext context, QLPreviewReply reply, out NSError error);
+	delegate NSData QLPreviewReplyDataCreationHandler (QLPreviewReply reply, out NSError error);
+	delegate CGPDFDocument QLPreviewReplyUIDocumentCreationHandler (QLPreviewReply reply, out NSError error);
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface QLPreviewReply
+	{
+		[Export ("stringEncoding")]
+		nuint StringEncoding { get; set; }
```
Comment (spouliot) [inline src/quicklookUI.cs:220]: from wiki / diff
```
+/** String encoding for text or html based previews. Defaults to NSUTF8StringEncoding. */
+@property NSStringEncoding stringEncoding;
```

there's already an `enum` for this, named `NSStringEncoding`
see `src/Foundation/Enum.cs`

**Review by spouliot — APPROVED** [review]

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
Please address the comments from @spouliot

**Review by rolfbjarne — APPROVED** [review]

File: src/quicklookUI.cs (hunk 1 above)
Comment (rachelkang) [reply src/quicklookUI.cs:183]: All your comments address things I also missed in QuickLook - I'll make a separate PR to correct those in that framework https://github.com/xamarin/xamarin-macios/pull/12337/files

**Review by tj-devel709 — APPROVED** [review]

File: src/quicklookUI.cs (hunk 1 above)
Comment (rachelkang) [reply src/quicklookUI.cs:183]: https://github.com/xamarin/xamarin-macios/pull/12402

**Comment by rachelkang** [comment]
test failure is unrelated https://github.com/xamarin/maccore/issues/2414

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**